plenum/bls/bls_key_register_pool_manager.py

```diff
@@ -21,12 +25,18 @@ def get_key_by_name(self, node_name, pool_state_root_hash=None):
         if self._current_pool_state_root_hash != pool_state_root_hash:
             self._current_pool_state_root_hash = pool_state_root_hash
             self._load_keys_for_root(pool_state_root_hash)
-
+        if not self._pool_manager.config.VALIDATE_SIGN_WITHOUT_BLS_KEY_PROOF and \
+                self._current_bls_keys_proof.get(node_name, None) in [False, None]:
+            logger.info("{} has no proof of possession for BLS public key.".format(node_name))
+            return None
         return self._current_bls_keys.get(node_name)
 
     def _load_keys_for_root(self, pool_state_root_hash):
         self._current_bls_keys = {}
+        self._current_bls_keys_proof = {}
         for data in self._pool_manager.reqHandler.get_all_node_data_for_root_hash(
                 pool_state_root_hash):
             if BLS_KEY in data:
                 self._current_bls_keys[data[ALIAS]] = data[BLS_KEY]
+                self._current_bls_keys_proof[data[ALIAS]] = \
```

Comment:
Why not doing like in BlsKeyRegisterPoolLedger where we fill `_current_bls_keys` with the keys for which we have proof only?

Reply:
Removed logic with _current_bls_keys_proof